// ngx-alerts/esm5/lib/ngx-alerts.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
var NgxAlertsComponent = /** @class */ (function () {
    function NgxAlertsComponent() {
    }
    /**
     * @return {?}
     */
    NgxAlertsComponent.prototype.ngOnInit = function () {
    };
    return NgxAlertsComponent;
}());
export { NgxAlertsComponent };
NgxAlertsComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-ngx-alerts',
                template: "\n    <p>\n      ngx-alerts works!\n    </p>\n  ",
                styles: []
            },] },
];
/** @nocollapse */
NgxAlertsComponent.ctorParameters = function () { return []; };
function NgxAlertsComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxAlertsComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxAlertsComponent.ctorParameters;
}
